composition/src/api_composition_cli.cpp

```diff
@@ -51,6 +51,9 @@ int main(int argc, char * argv[])
     rclcpp::executor::FutureReturnCode::SUCCESS)
   {
     fprintf(stderr, "api_composition_cli was interrupted. Exiting.\n");
+    if (!rclcpp::ok()) {
+      return 0;
+    }
```

Comment:
In the case where the future contains not success (and this script hasn't been asked to terminate using a SIGINT) I think this script should return a non-zero code.